app/src/main/java/com/example/meetup/LoginActivity.java

```diff
@@ -0,0 +1,148 @@
+package com.example.meetup;
+
+import androidx.appcompat.app.AlertDialog;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.net.ParseException;
+
+import android.app.ProgressDialog;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+import android.widget.Button;
+import android.widget.Toast;
+
+import com.google.android.gms.auth.api.signin.GoogleSignIn;
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
+import com.google.android.gms.auth.api.signin.GoogleSignInClient;
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions;
+import com.google.android.gms.common.SignInButton;
+import com.google.android.gms.common.api.ApiException;
+import com.google.android.gms.tasks.Task;
+import com.google.android.material.textfield.TextInputEditText;
+import com.parse.LogInCallback;
+import com.parse.ParseUser;
+
+import java.util.HashMap;
+import java.util.Objects;
+
+public class LoginActivity extends AppCompatActivity {
+
+    private TextInputEditText username;
+    private TextInputEditText password;
+    private Button login;
+    private Button navigateSignup;
+    private ProgressDialog progressDialog;
+    public static final int RC_SIGN_IN = 7;
+    GoogleSignInClient mGoogleSignInClient;
+    public static final String TAG = "LoginActivity";
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_login);
+
+        progressDialog = new ProgressDialog(LoginActivity.this);
+
+        username = findViewById(R.id.username);
+        password = findViewById(R.id.password);
+        login = findViewById(R.id.login);
+        navigateSignup = findViewById(R.id.navigatesignup);
+
+        login.setOnClickListener(v -> login(username.getText().toString(), password.getText().toString()));
+
+        navigateSignup.setOnClickListener(v -> {
+            startActivity(new Intent(LoginActivity.this, SignUpActivity.class));
+        });
+
+        setUpGoogleSignIn();
+        SignInButton googleSignIn = findViewById(R.id.sign_in_button);
+
+        Objects.requireNonNull(googleSignIn).setOnClickListener(v -> {
+            if (v.getId() == R.id.sign_in_button) {
+                signIn();
+            }
```

Comment:
Why do we need to check the ID of the view? Since we only set the listener to the Google sign in button, it can only be triggered with the Google sign in button is clicked.

Reply:
Agree. Objects.requireNonNull(googleSignIn) should be enough